cli/cmd/volume.go

```diff
@@ -202,13 +210,36 @@ var volumeGetCmd = &cobra.Command{
 	},
 }
 
+func volumeOptionJsonHandler(cmd *cobra.Command, volname string, options []string) error {
+	vopt := make(map[string]string)
+	for op,val := range options {
+		if op % 2 == 0 {
+			vopt[val] = options[op+1]
+		}else{
+			continue
+		}
+	}
+	var op = map[string]interface{}{"options": vopt}
+	opjson, _ := json.Marshal(op)
+	option := "'" + string(opjson) + "'"
+	err := client.VolumeSet(volname, option)
+	return err
+}
+
 var volumeSetCmd = &cobra.Command{
 	Use:   "set",
 	Short: helpVolumeSetCmd,
 	Run: func(cmd *cobra.Command, args []string) {
-		validateNArgs(cmd, 3, 3)
+		validateNArgs(cmd, 3, 0)
 		volname := cmd.Flags().Args()[0]
-		fmt.Println("SET:", volname)
+		options := os.Args[5:]
```

Comment:
Do not take Args directly from os.Args, This will have raw arguments which are not parsed. Use cmd.Flags().Args()

Reply:
I tried using cmd.Flags().Args()
but, like I explained earlier:
$./glustercli volume set testvol --data afr.eager-lock on foo bar
The volume options array being created is [on foo bar].
The first volume option entered is being ommitted.
Hence, thought of extracting the value of the flag: 'data' and then append it to the array.

Reply:
That is because of `--option` or `--data` flag. Both of them not required. Just use 

    $./glustercli volume set testvol afr.eager-lock on foo bar